### DISCUSSION TITLE
legend in front of grid

### DISCUSSION THREAD

**Discussion by patoutafay** [pyqtgraph/pyqtgraph#2688] [General]
it is possible to put the legend of a pyqtgraph in front of the grid in order to don't see the grid in the legend?

**Comment by pijyoi** [comment]
Grid lines are drawn by `AxisItem`, which has a Z-value of 0.5. So to achieve what you want, you would call LegendItem::setZValue(1)
https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/PlotItem/PlotItem.py#L331-L332

**Comment by patoutafay** [comment]
it seems to doesn't work. wahtever z value i use, grid is always on top of erveything

**Comment by pijyoi** [comment]
If you take a look at https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/examples/Legend.py, it demonstrates 2 ways of creating the `LegendItem`, the 1st way being commented out.

The 2nd way (together with setting z value) will allow you to place the legend in front of the grid lines.

According to the source code, the 1st way automatically sets the `ViewBox` to be the `ParentItem` of the `LegendItem`. However, `ViewBox` sets its own ZValue to be -100, thus all of `ViewBox`'s children would be placed behind the grid lines.
